runners/gearpump/src/main/java/org/apache/beam/runners/gearpump/GearpumpRunner.java

```diff
@@ -75,9 +74,10 @@ public static GearpumpRunner fromOptions(PipelineOptions options) {
 
   public <OutputT extends POutput, InputT extends PInput> OutputT apply(
       PTransform<InputT, OutputT> transform, InputT input) {
-    if (Window.Bound.class.equals(transform.getClass())) {
+    if (Window.Bound.class.equals(transform.getClass())
+        && isNullOrIdentityWindowFn(((Window.Bound) transform).getWindowFn())) {
```

Comment:
Does this change relate to the Gearpump upgrade? I feel like it is a separate concern. It is OK to bundle them in one PR, but would you mind splitting the commits?

```
git reset HEAD~
git add -p
# Choose just the Gearpump version changes
git commit -m 'Update Gearpump version'
git add -p
# Choose just the commits where you added the translator
git commit -m 'Add Window.Bound translator'
git add -p
# Choose just the other little changes
git commit -m 'Skip window assignment when windows don't change'
```

etc. It is up to you how many commits, but I would just like the Gearpump version bump to be isolated.